core-contracts/Dex/src/main/java/network/balanced/score/core/dex/AbstractDex.java

```diff
@@ -633,6 +634,18 @@ void exchange(Address fromToken, Address toToken, Address sender,
         totalTokensInPool.set(fromToken, newFromToken);
         totalTokensInPool.set(toToken, newToToken);
 
+        if (isSell) {
+            oldFromToken = newFromToken;
+            oldToToken = newToToken;
+
+            BigInteger newFromTokenWithBalnFee = oldFromToken.add(balnFees);
+            BigInteger newToTokenAfterFeeSwap = (oldFromToken.multiply(oldToToken)).divide(newFromTokenWithBalnFee);
+
+            balnFees = oldToToken.subtract(newToTokenAfterFeeSwap);
+
+            totalTokensInPool.set(fromToken, newFromTokenWithBalnFee);
+            totalTokensInPool.set(toToken, newToTokenAfterFeeSwap);
+        }
```

Comment:
```suggestion
        if (isSell) {
            BigInteger fromTokenWithoutFees = newFromToken;
            BigInteger toTokenWithoutFees = newToToken;

            newFromToken = fromTokenWithoutFees.add(balnFees);
            newToToken = (fromTokenWithoutFees.multiply(toTokenWithoutFees)).divide(newFromToken);

            balnFees = oldToToken.subtract(newToToken);
         }
        
         totalTokensInPool.set(fromToken, newFromToken);
         totalTokensInPool.set(toToken, newToToken);
```

Not sure if i wrote that completely correct but we can avoid the additional setter as well.